media_cdn/dualtoken.rb

```diff
@@ -0,0 +1,179 @@
+# Copyright 2023 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START mediacdn_dualtoken_sign_token]
+require "base64"
+require "time"
+require "openssl"
+require "ed25519"
+
+class Dualtoken
+  def base64_encoder value
+    #   Returns a base64-encoded string compatible with Media CDN.
+    #
+    #   Media CDN uses URL-safe base64 encoding and strips off the padding at the end.
+
+    encoded_str = Base64.urlsafe_encode64(value).encode("utf-8")
+    encoded_str.delete "="
```

Comment:
`encoded_str.delete_suffix! '='` is probably more efficient
Also `delete` is non destructive, so it's a noop?

Reply:
`encoded_str.delete_suffix! '='` cannot remove the trailing '='

needs to stick to `encoded_str.delete "="`

Reply:
I see. I took a look at the docs and looks like you can do

```ruby
encoded_str = Base64.urlsafe_encode64(value, padding:false)
```

Since `base64` returns `US-ASCII`, you don't need to encode it either.

Reply:
with padding:false, it works!

pushed a new version